src/viam/sdk/registry/registry.hpp

```diff
@@ -66,9 +67,11 @@ template <typename ResourceClientT,
 class ResourceRegistration2 : public ResourceRegistration {
    public:
     using ResourceRegistration::ResourceRegistration;
-    std::shared_ptr<ResourceServer> create_resource_server(
-        std::shared_ptr<ResourceManager> manager) override {
-        return std::make_shared<ResourceServerT>(manager);
+    std::shared_ptr<ResourceServer> create_resource_server(std::shared_ptr<ResourceManager> manager,
```

Comment:
Part of the reason moving `register_service` to `create_resource_server` will be nice is that if all resources start using `ResourceRegistration2` (or something similar) in [RSDK-3030](https://viam.atlassian.net/browse/RSDK-3030), this templated version of `create_resource_server` will be the only place we need to call `register_service`.